### PR TITLE
Chore: Update Slate to 0.47.8

### LINKED ISSUE #17430: Update Slate
If we are to invest more in Slate we really need to upgrade Slate

### PR DESCRIPTION
**Pull request by kaydelaney**
**What this PR does / why we need it**:
Updates slate to 0.47.8 and refactors code making use of slate accordingly.

**Which issue(s) this PR fixes**:
Closes #17430

### CODE REVIEW

**Review by davkal — APPROVED** [review]
LGTM, tested Prometheus, Loki, Azure Logs, Data links

@dprokop do you want to take another look too?

**Review by hugohaggmark — COMMENTED** [review]
Wow, what a huge undertaking! Great job with this.
I left some small comments and I found an unexpected behaviour when you select all the text using SHIFT+OPTION+LEFT arrow on MacOS all the text is first selected and the automatically erased.

unexpected-select-all

File: packages/grafana-ui/rollup.config.ts [hunk 1]
```diff
@@ -47,19 +47,20 @@ const buildCjsPackage = ({ env }) => {
...
             'Stack',
-            'Record',
           ],
+          'node_modules/immutable/dist/immutable.js': ['Record', 'Set', 'Map', 'List', 'OrderedSet', 'is', 'Stack'],
           '../../node_modules/esrever/esrever.js': ['reverse'],
         },
       }),
       resolve(),
-      sourceMaps(),
+      // sourceMaps(),
```
[the remark this answers is not in the document]
Comment (dprokop) [reply packages/grafana-ui/rollup.config.ts]: Let's leave it here for now and add a comment why this is actually commented out. I need to investigate it deeper at some point. Probably worth creating an issue too.

**Comment by kaydelaney** [comment]
@hugohaggmark Thanks for your review! Will look into that  unexpected behavior you mentioned.

**Comment by dprokop** [comment]
@kaydelaney minor UX observation in DataLinks - pressing Backspace closes suggestions menu but does not remove character from the input

File: public/app/features/explore/slate-plugins/clear.test.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import Plain from 'slate-plain-serializer';
+import React from 'react';
+import { Editor } from '@grafana/slate-react';
```
Comment (dprokop) [inline public/app/features/explore/slate-plugins/clear.test.tsx]: What's the plan for maintaining this vs. eventually going back to non-forked version?

**Review by dprokop — APPROVED** [review]
Really solid and impressive work @kaydelaney!

File: public/app/features/explore/slate-plugins/clear.test.tsx (hunk 2 above)
Comment (kaydelaney) [reply public/app/features/explore/slate-plugins/clear.test.tsx]: I think we can switch back to the non-forked version as soon as my fixes get merged, but unfortunately things move fairly slowly upstream

**Comment by kaydelaney** [comment]
Fixed the issues mentioned by @hugohaggmark and @dprokop. Should be okay to merge! Thoughts @davkal?

**Comment by torkelo** [comment]
great work on this tricky upgrade! Hopefully, next upgrade won't require as much work :)

**Pull request merged**